lua/autorun/server/sv_autoclean.lua

```diff
@@ -1,36 +1,57 @@
-	CreateConVar('cfc_autoclean','500',{FCVAR_ARCHIVE, FCVAR_NOTIFY },'Autocleans the server based on seconds given')
-	
+
+local ConVarFlags = {FCVAR_ARCHIVE, FCVAR_NOTIFY}
+local DefaultCleanIntervalInSeconds = "500"
+CreateConVar("cfc_autoclean", DefaultCleanIntervalInSeconds, ConVarFlags, "Autocleans the server based on seconds given")
+
+local BlacklistedEntityTypes = {
+    ["gmod_wire_turret"] = true,
+    ["gmod_turret"] = true
+}
+
+local CleanupCommands = {
+    ["r_cleardecals"] = true,
+    ["stopsound"] = true
+}
+
+local function notifyPlayers( count )
+    local Message = "[CFC_Autoclean] Removed " .. tostring(count) .. "objects."
+
+    print(Message) -- and server
+    for _, ply in pairs( player.GetHumans() ) do
+        ply:ChatPrint(Message)
+    end
+end
+
+local function runCleanupCommandsOnPlayers()
+    for _, ply in pairs( player.GetHumans() ) do
+        for command, _ in pairs( CleanupCommands ) do
+            ply:ConCommand( command )
+        end
+
+        ply:ChatPrint("[CFC_Autoclean] Cleaning server...")
+    end
+end
+
 function cfcCleanServer()
-	local player_weps = {}
-	local turrets = table.Add(ents.FindByClass("gmod_wire_turret"),ents.FindByClass("gmod_turret"))
-	local count = 0
-	--Manual fix for removing weapons held by player
-	for _ , v in pairs(player.GetHumans()) do
-		v:ChatPrint('[CFC_Autoclean] Cleaning server..')
-		--Also cleanup decals and sounds while we're here
-		v:ConCommand('r_cleardecals')
-		v:ConCommand('stopsound')
-
-		--
-		local wp = v:GetWeapons()
-		for _,m in pairs(wp) do
-			if (not table.HasValue(player_weps, m))then
-				table.insert(player_weps, m)
-			end
-		end
-	end
-	
-	for k, v in pairs( ents.GetAll() ) do
-			if v:IsWeapon() and not (table.HasValue(player_weps,v) or (table.HasValue(turrets,v))) then
-				count = count + 1
-				v:Remove()
-			end
-	end
-	
-	MsgAll('[CFC_Autoclean] Removed ' .. tostring(count) .. ' objects.')
+    local removedCount = 0
+    
+    runCleanupCommandsOnPlayers()
+
+    for _, entity in pairs( ents.GetAll() ) do
+        if not IsValid( entity ) then continue end
+
+        local isEquippedWeapon = entity:IsWeapon() and IsValid( entity.Owner )
+        local classIsBlacklisted = BlacklistedEntityTypes[entity:GetClass()]
+        if isEquippedWeapon or not classIsBlacklisted then continue end
+
+        removedCount = removedCount + 1
+        entity:Remove()
+    end
+    
+    MsgAll("[CFC_Autoclean] Removed " .. tostring( removedCount ) .. " objects.")
 end
 
-timer.Create('cfcautoclean',GetConVar('cfc_autoclean'):GetInt(),0,cfcCleanServer)
+timer.Create("cfc_autoclean", GetConVar("cfc_autoclean"):GetInt(), 0, cfcCleanServer)
```

Comment:
Following convention:

```suggestion
timer.Create("cfc_AutoClean", GetConVar("cfc_autoclean"):GetInt(), 0, cfcCleanServer)
```